### ISSUE TITLE
weather@mockturtl errors in .xsession.log

### ISSUE THREAD

**Issue by ChooChooAl** [linuxmint/cinnamon-spices-applets#3421]
Version 2.6.9 (2020-12-25 11:37:53)
Cinnamon 4.8.5
Mint 20.1 Cinnamon 64bit
Intel built in graphics
 * Snippet or attachment of ~/.xsession-errors
(cinnamon:7518): Cjs-CRITICAL **: 10:48:44.639: JS ERROR: Gio.IOErrorEnum: Error opening file /home/alan/.config/weather-mockturtl/locations.json: No such file or directory
LoadContents/</<@/home/alan/.local/share/cinnamon/applets/weather@mockturtl/3.8/applet.js:1898:47

 ```
@mockturtl

Getting a few of these errors in .xsession.log
I'm trying to minimize the activity to my HD.   There aren't many but I thought I'd report the issue in case something might have been missed in the code.

**Comment by Gr3q** [comment]
No you are right, it supposed to check if the file exists before, but I guess it passes that.

Just to double check, you don't have that file on said location, right?

**Comment by ChooChooAl** [comment]
.config/weather-mockturtl/locations.json 
does not exist.  Not even the folder weather-mockturtl

**Comment by Gr3q** [comment]
That's expected, thank you. I'll look into it. This is only called once when the applet is created so it shouldn't put extra strain on your HDD, but I'll try to make sure it never fails like this

**Comment by ChooChooAl** [comment]
Thanks.  Low priority of course.    Thanks.
Ironically Mint Cin. 2.0 had tons of errors in the .xsession-errors file, tons.  But things were fixed in 20.1 now and the load on that file is way down.  You can actually see errors now LOL.

**Comment by Gr3q** [comment]
There were some problems with the IO code (obviously), I've fixed them, also using more of the async functions so they are not blocking anymore. 

Will be in the next release
